packages/integrations/vercel/src/static/adapter.ts

```diff
@@ -118,6 +118,11 @@ export default function vercelStatic({
 							continue: true,
 						},
 						{ handle: 'filesystem' },
+						...routes.find(route => route.pathname === "/404") ? [{
```

Comment:
I think the safest way to handle this case is by using the `.pattern` property. Also, this check doesn't take into consideration the `base` option, so we can't redirect to `/404.html` because it could be outside of Astro jurisdiction

Reply:
`pattern`, after being converted from RegExp to a string, would be the most difficult to match against because it includes base and trailingSlash.

To clarify the intent here, we only need to know whether there will be a 404.html in the filesystem.